enumerables.rb

```diff
@@ -0,0 +1,96 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    if is_a? Hash
+      length.times { |element| yield(keys[element], self[keys[element]]) }
+    else
+      length.times { |element| yield(self[element]) }
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    length.times { |element| yield(self[element], element) }
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each { |element| new_array.push(element) if yield(element) }
+    new_array
+  end
+
+  def my_all?(*par)
+    condition = true
+    if !par[0].nil?
+      my_each { |element| condition = false unless par[0] === element } # rubocop:disable Style/CaseEquality
+    elsif !block_given?
+      my_each { |element| condition = false unless element }
+    else
+      my_each { |element| condition = false unless yield(element) }
+    end
+    condition
+  end
+
+  def my_any?(*par)
+    condition = false
+    if !par[0].nil?
+      my_each { |element| condition = true if par[0] === element } # rubocop:disable Style/CaseEquality
+    elsif !block_given?
+      my_each { |element| condition = true if element }
+    else
+      my_each { |element| condition = true if yield(element) }
+    end
+    condition
+  end
+
+  def my_none?(*par)
+    condition = true
+    if !par[0].nil?
+      my_each { |element| condition = false if par[0] === element } # rubocop:disable Style/CaseEquality
+    elsif !block_given?
+      my_each { |element| condition = false if element }
+    else
+      my_each { |element| condition = false if yield(element) }
+    end
+    condition
+  end
+
+  def my_count(par = nil)
+    index = 0
+    if block_given?
+      my_each { |element| index += 1 if yield(element) }
+    elsif par
+      my_each { |element| index += 1 if element == par }
+    else
+      index = length
+    end
+    index
+  end
+
+  def my_map(proc = nil)
+    return to_enum(:my_map) unless block_given?
+
+    new_array = []
+    my_each { |element| new_array.push(proc.nil? ? yield(element) : proc.call(element)) }
+    new_array
+  end
+
+  def my_inject(*par)
+    memo = par[0] if par[0].is_a? Integer
+    if par[0].is_a?(Symbol) || par[0].is_a?(String)
+      my_each { |element| memo = memo ? memo.send(par[0], element) : element }
+      memo
+    else
+      my_each { |element| memo = memo ? yield(memo, element) : element }
+    end
+    memo
+  end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving a NameError
![image](https://user-images.githubusercontent.com/52209067/79671469-a5c5e780-8187-11ea-9d06-f291c53d4f44.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(2, :+) == [2, 5, 6].inject(2, :+) => 15
```
- Yours is retrieving a LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/79671502-e0c81b00-8187-11ea-8afc-64d07788e63f.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/79671514-f0dffa80-8187-11ea-8a4f-27fae3f6e849.png)

Reply:
fixed

Reply:
my_inject method working now